test/shared/poolUtilities.ts

```diff
@@ -0,0 +1,178 @@
+import { BigNumber, BigNumberish, constants, Contract, ContractTransaction, utils, Wallet } from 'ethers'
+import { TestUniswapV3Callee } from '../../typechain/TestUniswapV3Callee'
+import { TestERC20 } from '../../typechain/TestERC20'
+import { IUniswapV3Pool } from '../../typechain'
+import { MockProvider } from 'ethereum-waffle'
+
+interface SwapExact0For1TestCase {
+  zeroForOne: true
+  exactOut: false
+  amount0: BigNumberish
+  sqrtPriceLimit?: BigNumber
+  increaseTime: number
+}
+
+interface SwapExact1For0TestCase {
+  zeroForOne: false
+  exactOut: false
+  amount1: BigNumberish
+  sqrtPriceLimit?: BigNumber
+  increaseTime: number
+}
+
+interface Position {
+  tickLower: number
+  tickUpper: number
+  liquidity: BigNumberish
+}
+
+export interface PoolTestCase {
+  description: string
+  feeAmount: number
+  tickSpacing: number
+  startingPrice: BigNumber
+  positions: Position[]
+  swapTests: SwapTestCase[]
+}
+
+export type SwapTestCase = SwapExact0For1TestCase | SwapExact1For0TestCase
+
+export const MIN_SQRT_RATIO = BigNumber.from('4295128739')
+export const MAX_SQRT_RATIO = BigNumber.from('1461446703485210103287273052203988822378723970342')
+
+export function getCreate2Address(
```

Comment:
already have this https://github.com/Uniswap/uniswap-v3-periphery/blob/main/test/shared/computePoolAddress.ts